### ISSUE TITLE
No verification of website

### ISSUE THREAD

**Issue by MaSven** [62726164/ed25519-login#2]
With this i don't know who the website is. So this is easy phishable as i can just forward the signature.

**Comment by 62726164** [comment]
I plan to switch to signing a nonce rather than the current time. Only the website and the user would know the nonce. A fake nonce would not be recognized by the real website.

**Comment by intob** [comment]
An alternative would be appending the unix timestamp with the host domain. This limits the scope of the signature to the scope of the domain (or subdomain). 

In a recent project, I switched from nonce to timestamp signing for two reasons:
- Eliminates a round-trip to request the nonce
- Server must not remember any nonce for the client (completely stateless)

You also don't need to store the whole public key (if you don't want/need), just a hash of it. The client can then send the pub key with the signature, and you can verify that it matches your stored hash.

Have fun :)

**Comment by 62726164** [comment]
Yes, adding the domain to the timestamp is another good idea to limit the scope. 

Thanks for the comment and the link.
